### ISSUE TITLE
New Rule Suggestion: Allow/disallow an empty newline after var statement

### ISSUE THREAD

**Issue by g13n** [eslint/eslint#2057]
As of today there is no consistency in separating variable declarations from the rest of the code. Some developers leave an empty line between `var` statements and the rest of the code like:

        var foo;

        // do something with foo

Whereas others don't leave any empty newlines at all.

        var foo;
        // do something with foo

The problem is when these developers work together in a project. I suggest a new rule to make this pattern consistent across the project.

**Comment by nzakas** [comment]
Sure, if you want to implement, happy to review.

**Comment by gyandeeps** [comment]
I like this rule as it separates the declarations from implementation logic visually. It becomes easy to read.
What happens if this is the code statement:

```.js
function test(){
    var a = 1;
    var b = 2;

    return a + b;
}
```

My expectation: It should not throw an error.

**Comment by g13n** [comment]
Thanks @nzakas.

@gyandeeps it works (doesn't throw an error.)

That's the [Google style code (with multiple variables)](https://github.com/g13n/eslint/blob/4d91e061022fda9bfd8dec246812cd029c9086d8/lib/rules/newline-after-var.js#L29) and I have [test cases to test that style](https://github.com/g13n/eslint/blob/4d91e061022fda9bfd8dec246812cd029c9086d8/tests/lib/rules/newline-after-var.js#L26).
